src/orion/core/__init__.py

```diff
@@ -294,6 +294,18 @@ def define_worker_config(config):
         ),
     )
 
+    worker_config.add_option(
```

Comment:
`reservation_timeout` above should be deprecated like the `max_idle_time` above. :sweat_smile: We could set deprecation version to v0.4.0.

Reply:
done